src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/UriExtensions.cs

```diff
@@ -14,8 +15,19 @@ internal static class UriExtensions
     ///  Converts the specified <see cref="Uri"/> into a file path that matches
     ///  a Roslyn <see cref="TextDocument.FilePath"/>.
     /// </summary>
-    public static string GetDocumentFilePath(this Uri uri)
-        => RazorUri.GetDocumentFilePathFromUri(uri);
+    public static string GetDocumentFilePath(this DocumentUri uri)
+        => RazorUri.GetDocumentFilePathFromUri(uri.GetRequiredParsedUri());
+
+    public static string GetAbsoluteOrUNCPath(this DocumentUri uri)
+    {
+        if (uri is null)
+        {
+            throw new ArgumentNullException(nameof(uri));
+        }
```

Comment:
Is this necessary in a nullable annotated file? If so, we have an `ArgHelper.ThrowIfNull` helper method